swap/swap.go

```diff
@@ -120,74 +119,97 @@ func swapRotatingFileHandler(logdir string) (log.Handler, error) {
 	}
+	// connect to the backend
 	backend, err := ethclient.Dial(backendURL)
 	if err != nil {
-		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %s", backendURL, err)
+		return nil, fmt.Errorf("error connecting to Ethereum API %s: %v", backendURL, err)
 	}
-
-	// we may not need this check, and we could maybe even get rid of this constant completely
-	if params != nil && params.InitialDepositAmount == 0 {
-		// need to prompt user for initial deposit amount
-		// if 0, can not cash in cheques
-		prompter := console.Stdin
-
-		// ask user for input
-		input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment: ")
-		if err != nil {
-			return nil, err
-		}
-		// check input
-		val, err := strconv.ParseInt(input, 10, 64)
-		if err != nil {
-			// maybe we should provide a fallback here? A bad input results in stopping the boot
-			return nil, fmt.Errorf("Conversion error while reading user input: %v", err)
-		}
-		log.Info("Chequebook initial deposit amount: ", "amount", val)
-		params.InitialDepositAmount = uint64(val)
+	// get the networkID of the backend
+	var networkID *big.Int
+	if networkID, err = backend.NetworkID(context.TODO()); err != nil {
+		return nil, fmt.Errorf("error retrieving networkID from backendURL: %v", err)
 	}
-
-	swap := new(
+	// verify that we have not used SWAP before on a different networkID
+	if err := checkNetworkID(networkID.Uint64(), stateStore); err != nil {
+		return nil, err
+	}
+	auditLog.Info("SWAP initialized on backend network ID", "ID", networkID.Uint64())
+	// create the owner of SWAP
+	owner := createOwner(prvkey)
+	// start the chequebook
+	var contract contract.Contract
+	if contract, err = StartChequebook(chequebookAddressFlag, initialDepositAmountFlag, stateStore, owner, backend); err != nil {
+		return nil, err
+	}
+	// create the SWAP instance
+	return new(
 		stateStore,
-		prvkey,
+		owner,
 		backend,
-		params)
-
-	return swap, nil
+		params,
+		contract,
+	), nil
 }
 
 const (
-	balancePrefix        = "balance_"
-	sentChequePrefix     = "sent_cheque_"
-	receivedChequePrefix = "received_cheque_"
+	balancePrefix          = "balance_"
+	sentChequePrefix       = "sent_cheque_"
+	receivedChequePrefix   = "received_cheque_"
+	connectedChequebookKey = "connected_chequebook"
+	connectedBlockchainKey = "connected_blockchain"
 )
 
+// checkNetworkID verifies whether we have initialized SWAP before and ensures that we are on the same backendNetworkID if this is the case
+func checkNetworkID(currentNetworkID uint64, s state.Store) (err error) {
+	var connectedBlockchain uint64
+	err = s.Get(connectedBlockchainKey, &connectedBlockchain)
+	// error reading from database
+	if err != nil && err != state.ErrNotFound {
+		return fmt.Errorf("error querying usedBeforeAtNetwork from statestore: %v", err)
+	}
+	// initialized before, but on a different networkID
+	if connectedBlockchain != currentNetworkID {
```

Comment:
This seems to be broken now, I can't start swarm anymore. If the datadir was never used before (leading to `err` being `state.ErrNotFound `) then `connectedBlockchain ` will be 0 an this check will always fail.

Reply:
Fixed